source/ios_net/source/main.c

```diff
@@ -139,19 +294,41 @@ void writeData(void* data, uint32_t size)
         }
     }
 
+    int interface_id;
+    if (interface[0] == 'U' && interface[1] == 'S' && interface[2] == 'B') {
+        interface_id = 0;
+    }
+    else { //if (interface[0] == 'w' && interface[1] == 'l') {
+        interface_id = 1;
+    }
+
     uint64_t time;
     IOS_GetAbsTime64(&time);
 
-    pcaprec_hdr_t rec_hdr = { 0 };
-    rec_hdr.ts_sec = time / 1000000;
-    rec_hdr.ts_usec = time - (rec_hdr.ts_sec * 1000000);
-    rec_hdr.orig_len = size;
-    rec_hdr.incl_len = (size >= _configuration.maxPacketSize) ? _configuration.maxPacketSize : size;
+    pcapng_epb_t ebp_hdr = { 0 };
+    ebp_hdr.block_type = 6;
+    ebp_hdr.interface_id = interface_id;
+    ebp_hdr.timestamp = time; // this is wrong, but I don't think I care
```

Comment:
What is this comment referring to?

Reply:
I believe it is either the wrong units (usec vs. nsec vs. msec) or the epoch is wrong, I can't remember.

Reply:
The timestamp in enhanced packet blocks is made up of 2 sections, a uint32 high and uint32 low. These combined make up the time in units since the standard unix epoch. However the definition of "unit" depends on the `if_tsresol` option set on the interface. See https://www.ietf.org/archive/id/draft-tuexen-opsawg-pcapng-03.html#name-enhanced-packet-block for details on the pcapng format. This is the source I used when writing our pcapng parser, which implements the decoding of this value (for reference) https://github.com/PretendoNetwork/nex-viewer/blob/933c66cd346498085ef8e2c56aca9b993d88c34e/src/pcapng-parser.ts#L175-L188

https://www.ietf.org/archive/id/draft-tuexen-opsawg-pcapng-03.html#section-4.3-6.4
> Timestamp (High) and Timestamp (Low): upper 32 bits and lower 32 bits of a 64-bit timestamp. The timestamp is a single 64-bit unsigned integer that represents the number of units of time that have elapsed since 1970-01-01 00:00:00 UTC. The length of a unit of time is specified by the 'if_tsresol' option (see [Figure 10](https://www.ietf.org/archive/id/draft-tuexen-opsawg-pcapng-03.html#format_idb)) of the Interface Description Block referenced by this packet. Note that, unlike timestamps in the libpcap file format, timestamps in Enhanced Packet Blocks are not saved as two 32-bit values that represent the seconds and microseconds that have elapsed since 1970-01-01 00:00:00 UTC. Timestamps in Enhanced Packet Blocks are saved as two 32-bit words that represent the upper and lower 32 bits of a single 64-bit quantity.

https://www.ietf.org/archive/id/draft-tuexen-opsawg-pcapng-03.html#section-4.2-24.2.1
> The if_tsresol option identifies the resolution of timestamps. If the Most Significant Bit is equal to zero, the remaining bits indicates the resolution of the timestamp as a negative power of 10 (e.g. 6 means microsecond resolution, timestamps are the number of microseconds since 1970-01-01 00:00:00 UTC). If the Most Significant Bit is equal to one, the remaining bits indicates the resolution as negative power of 2 (e.g. 10 means 1/1024 of second). If this option is not present, a resolution of 10^-6 is assumed (i.e. timestamps have the same resolution of the standard 'libpcap' timestamps).